### PR TITLE
【Prim】Optimize composite rule by making scalar shape as 1

### PR DESCRIPTION
**Pull request by JiabinYang**
### PR types

Performance optimization
### PR changes

Others
### Describe

Pcard-66975
This PR Optimize prim performance by remove additional full shape

### CODE REVIEW

**Review by xiaoguoguo626807 — APPROVED** [review]
LGTM

File: python/paddle/incubate/autograd/composite_rules.py [hunk 1]
```diff
@@ -188,12 +190,13 @@ def gelu_composite(x, approximate):
         0.70710678118654752440  # /* 1/sqrt(2) */ copy from gelu-kernel.cc
     )
     M_2_SQRTPI = 1.12837916709551257390  # /* 2/sqrt(pi) */
-    one = ones(x.shape, x.dtype)
-    half = full(x.shape, 0.5, x.dtype)
+    full_shape = x.shape if len(x.shape) == 0 else [1]
+    one = ones(full_shape, x.dtype)
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py:194]: len(x.shape) == 0 this condition won't exist, better to check other same places

Comment (JiabinYang) [reply python/paddle/incubate/autograd/composite_rules.py:194]: 0D cases

Comment (JiabinYang) [reply python/paddle/incubate/autograd/composite_rules.py:194]: 0D cases

**Review by cyber-pioneer — APPROVED** [review]
LGTM

**Pull request merged**